src/librustc_driver/driver.rs

```diff
@@ -1019,6 +1019,7 @@ pub fn phase_4_translate_to_llvm<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
 
         passes.push_pass(box mir::transform::erase_regions::EraseRegions);
 
+        passes.push_pass(box mir::transform::instcombine::InstCombine::new());
         passes.push_pass(box mir::transform::add_call_guards::AddCallGuards);
         passes.push_pass(box borrowck::ElaborateDrops);
```

Comment:
You only need to reorder the pass to go after this line, which should be separated and commented.
This is the point where moves stop existing and you only have raw semantic-less copies.